astroid/nodes/scoped_nodes.py

```diff
@@ -484,6 +484,7 @@ def __init__(
         package=None,
         parent=None,
         pure_python=True,
+        doc_node=None,
```

Comment:
Please add typing to this parameter here, like it has been done for `path`. The typing in the docstring can be removed as we're moving to using typing within the code.
Same goes for the other occurrences of this pattern.